strings/base_coroutine_threadpool.h

```diff
@@ -202,6 +203,7 @@ namespace winrt::impl
     template <typename T>
     decltype(auto) get_awaiter(T&& value) noexcept
     {
+#ifdef WINRT_IMPL_COROUTINES
```

Comment:
I think we don't need this `#ifdef` any more. `has_co_await_member` and `has_co_await_free` will always be false if `WINRT_IMPL_COROUTINES` is not defined, so we fall through to the last case, which requires the type to be its own awaiter (and if we move the `find_awaitable_member` outside the `#ifdef` it will work), it will look for an `await_ready` which is fine.

Reply:
Unfortunately, Clang currently looks inside the `if constexpr` regardless and fails to compile.

Reply:
Oh, yeah, forgot about that. Even though it's `if constexpr`'d out, it still has to compile.